src/builder.rs

```diff
@@ -80,90 +80,94 @@ impl SearchBuilder {
     /// use rust_search::SearchBuilder;
     ///
     /// let search: Vec<String> = SearchBuilder::default()
-    /// .ext(".rs")
-    /// .build()
-    /// .collect();
+    ///     .ext(".rs")
+    ///     .build()
+    ///     .collect();
     /// ```
     pub fn ext(mut self, ext: impl Into<String>) -> Self {
         self.file_ext = Some(ext.into());
         self
     }
 
-    /// Set the depth to search to.
+    /// Set the depth to search to, meaning how many subdirectories to search in.
     /// ### Arguments
     /// * `depth` - The depth to search to.
     /// ### Examples
     /// ```rust
     /// use rust_search::SearchBuilder;
     ///
     /// let search: Vec<String> = SearchBuilder::default()
-    /// .depth(1)
-    /// .build()
-    /// .collect();
+    ///     .depth(1)
+    ///     .build()
+    ///     .collect();
     /// ```
     pub fn depth(mut self, depth: usize) -> Self {
         self.depth = Some(depth);
         self
     }
 
-    /// Searches for exact match
+    /// Searches for exact match.
+    ///
+    /// For example, if the search input is "Search", the file "Search.rs" will be found, but not "Searcher.rs".
     /// ### Examples
     /// ```rust
     /// use rust_search::SearchBuilder;
     ///
     /// let search: Vec<String> = SearchBuilder::default()
-    /// .search_input("name")
-    /// .strict()
-    /// .build()
-    /// .collect();
+    ///     .search_input("name")
+    ///     .strict()
+    ///     .build()
+    ///     .collect();
     /// ```
     pub fn strict(mut self) -> Self {
         self.strict = true;
         self
     }
 
     /// Set search option to be case insensitive.
+    ///
+    /// For example, if the search input is "Search", the file "search.rs" will be found.
     /// ### Examples
     /// ```rust
     /// use rust_search::SearchBuilder;
     ///
     /// let search: Vec<String> = SearchBuilder::default()
-    /// .search_input("name")
-    /// .ignore_case()
-    /// .build()
-    /// .collect();
+    ///     .search_input("name")
+    ///     .ignore_case()
+    ///     .build()
+    ///     .collect();
     /// ```
     pub fn ignore_case(mut self) -> Self {
         self.ignore_case = true;
         self
     }
 
-    /// Searches for hidden files
+    /// Searches for hidden files, meaning files that start with a dot.
```

Comment:
```suggestion
    /// Searches for hidden files.
```

I don't think that only files which have a dot are hidden...